visualdl/logic/sdk.cc

```diff
@@ -313,6 +340,32 @@ HistogramRecord<T> HistogramReader<T>::record(int i) {
   return HistogramRecord<T>(timestamp, step, left, right, std::move(frequency));
 }
 
+template <typename T>
+std::vector<HistogramRecord<T>> HistogramReader<T>::records() const{
+  std::vector<HistogramRecord<T>> res;
+  for (size_t i=0; i<reader_.total_records(); ++i){
+    res.push_back(record(i));
+  }
+
+  return res;
+}
+
+template <typename T>
+std::vector<HistogramRecord<T>> HistogramReader<T>::records_addi(size_t start_index) const{
+  std::vector<HistogramRecord<T>> res;
+
+  for (size_t i=start_index; i<reader_.total_records(); ++i){
```

Comment:
Coding style not correct, it should be 
` i = start_index; i < reader_.total_records();`